### ISSUE TITLE
Support updating compute instance metadata with separate resource

### ISSUE THREAD

**Issue by anilarora** [oracle/terraform-provider-oci#947]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

There are cases where we need to attach/update metadata for a compute instance, but cannot do so at the time of creation.  A concrete example is volume attachment.

The volume attachment resource has a dependency on the compute instance id, so it can only be created after the compute instance has been created.  However, to initialize the actual block volume, the compute instance needs the volume attachment metadata.  This forces us to use the remote-exec provisioner to execute the volume initialization code only after the attachment resource has been created.  This itself is problematic if the compute instance is on a private IP address and not accessible from the environment running terraform.

This feature request looks to add a new resource which can update compute metadata farther in the process.  Scripts executing during cloud-init can just simply loop and wait until the metadata is actually available before continuing with an initialization code.

### New or Affected Resource(s)

oci_core_instance_metadata

### Potential Terraform Configuration

```hcl
resource "oci_core_instance_metadata" "data_volume_info" {

  instance_ocid = "${oci_core_instance.myinstance.id}"

  metadata {
    data_volume_block_iqn = "${oci_core_volume_attachment.data_volume.iqn}"
    data_volume_block_ipv4 = "${oci_core_volume_attachment.data_volume.ipv4}"
    data_volume_block_port = "${oci_core_volume_attachment.data_volume.port}"
  }
}
```

**Comment by ravinitp** [comment]
We are very sorry that we couldn't respond to each and every issue reported on GitHub. 
Although we have refined the process to prioritize customer issues on GitHub, since this issue was reported a while ago, there is a good chance it may have been fixed in the latest version of Terraform Provider OCI. 
If you are still experiencing this issue, please create a new issue and label it as Bug.
